pkg/controller/infinispan/resources/secret/secret_controller.go

```diff
@@ -250,7 +250,7 @@ func (s secretResource) reconcileAdminSecret() error {
 }
 
 func (s secretResource) addCliProperties(secret *corev1.Secret, password string) {
-	service := s.infinispan.GetServiceName()
+	service := s.infinispan.GetAdminServiceName()
```

Comment:
@gustavonalle we should ensure that Secret will update this entry during update, otherwise Batch will nor work after that

Reply:
I suppose the batch after upgrade scenario is not tested currently?

Reply:
No upgrade tests in CI now. But seems to new CLI will be patched carefully in the `reconcileAdminSecret`

Reply:
You can run upgrade test with `make`
https://github.com/infinispan/infinispan-operator/blob/fba24c178eab6ed9bbcc3761530f08bd465fa67a/Makefile#L147-L166

https://github.com/infinispan/infinispan-operator/blob/fba24c178eab6ed9bbcc3761530f08bd465fa67a/Makefile#L168-L178

and keep cluster running with CLEANUP_INFINISPAN_ON_FINISH=false option. The one of the success condition for upgrade is Infinispan Server image tags should be different between branches or redefined with env variable

Reply:
> You can run upgrade test with make

Thanks, yeah, I was looking at those tests, but can I reproduce the secret issue you mentioned with it?

I tried running from my ```service``` branch, with:

```
make KUBECONFIG=... upgrade-test-local FROM_UPGRADE_VERSION=main
```

But it never does the upgrade, timing out in the second step, I guess 

> The one of the success condition for upgrade is Infinispan Server image tags should be different between branches or redefined with env variable

Is this a manual step?

Reply:
You can redefine target image with TO_OPERATOR_IMAGE env variables with the different server image tag
But seems to `build/upgrade-test-local.sh` need to be fixed a bit for the `main`

Reply:
ok, makes sense, in any case, the batch is not tested after the upgrade. I will add it to the test

Reply:
btw, tried with other branches and it failed too:

```
make KUBECONFIG=... upgrade-test-local FROM_UPGRADE_VERSION=2.1.4.Final TO_UPGRADE_VERSION=2.1.5.Final CLEANUP_INFINISPAN_ON_FINISH=false
```

Reply:
Server images for the `2.1.4.Final` and `2.1.5.Final` are the same, i.e. `quay.io/infinispan/server:12.1`. Upgrade will not trigger and required conditions flow will not pass.

Reply:
I thought the CSV file in each tag contained the image version for each operator:

https://github.com/infinispan/infinispan-operator/blob/2.1.4.Final/deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml#L312

https://github.com/infinispan/infinispan-operator/blob/2.1.5.Final/deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml#L313

Reply:
So the code uses 12.1 that is a "movable" tag, always points to the latest 12.1.x?

Reply:
Yes. correct. It get from the operator deployment yaml file. We discussed setting exact image version on update in the deployment here #1171 but images the same for the rest of the versions

Reply:
ouch. IMHO, having a single "movable" tag like ```latest``` is fine, but making every tag a movable tag kinds of defeat the purpose of a tag :smile:

Reply:
Yes, agree. But for full upgrade it's necessary to keep tags up to date in many branches. Seems to be too late now and can be customized with env variable only.

Reply:
I added a test and wasn't able to reproduce the secret issue @dmvolod, could you clarify the scenario?

Reply:
@gustavonalle looks good. can you try to see that upgraded cluster is wellFormed?

Reply:
This is already checked here I reckon:

https://github.com/infinispan/infinispan-operator/blob/fba24c178eab6ed9bbcc3761530f08bd465fa67a/test/e2e/main/main_test.go#L64-L66

Reply:
https://github.com/infinispan/infinispan-operator/blob/fba24c178eab6ed9bbcc3761530f08bd465fa67a/test/e2e/utils/constants.go#L44